scoring/run_workloads.py

```diff
@@ -0,0 +1,206 @@
+flags.DEFINE_integer('hparam_end_index', None, 'End index for tuning trials.')
+flags.DEFINE_integer('seed', None, 'Random seed for scoring.')
+flags.DEFINE_integer('submission_id',
+                     0,
+                     'Submission ID to generate study and hparam seeds.')
+flags.DEFINE_string('held_out_workloads_config_path',
+                    None,
+                    'Path to config containing held-out workloads')
+
+FLAGS = flags.FLAGS
+
+DATASETS = ['imagenet', 'fastmri', 'ogbg', 'wmt', 'librispeech', 'criteo1tb']
+
+WORKLOADS = {
+    'imagenet_resnet': {'max_steps': 186_666, 'dataset': 'imagenet'},
+    'imagenet_vit': {'max_steps': 186_666, 'dataset': 'imagenet'},
+    'fastmri': {'max_steps': 36_189, 'dataset': 'fastmri'},
+    'ogbg': {'max_steps': 80_000, 'dataset': 'ogbg'},
+    'wmt': {'max_steps': 133_333, 'dataset': 'wmt'},
+    'librispeech_deepspeech': {'max_steps': 48_000, 'dataset': 'librispeech'},
+    'criteo1tb': {'max_steps': 10_666, 'dataset': 'criteo1tb'},
+    'librispeech_conformer': {'max_steps': 80_000, 'dataset': 'librispeech'},
+}
+
+
+def read_held_out_workloads(filename):
+  with open(filename, "r") as f:
+    held_out_workloads = json.load(f)
+  return held_out_workloads
+
+
+def container_running():
+  docker_client = docker.from_env()
+  containers = docker_client.containers.list()
+  if len(containers) == 0:
+    return False
+  else:
+    return True
+
+
+def wait_until_container_not_running(sleep_interval=5 * 60):
+  while container_running():
+    time.sleep(sleep_interval)
+  return
+
+
+def main(_):
+  framework = FLAGS.framework
+  run_fraction = FLAGS.run_percentage / 100.
+  experiment_name = FLAGS.experiment_name
+  docker_image_url = FLAGS.docker_image_url
+  submission_path = FLAGS.submission_path
+  tuning_search_space = FLAGS.tuning_search_space
+  num_studies = FLAGS.num_studies
+  num_tuning_trials = FLAGS.num_tuning_trials
+  hparam_start_index_flag = ''
+  hparam_end_index_flag = ''
+  if FLAGS.hparam_start_index:
+    hparam_start_index_flag = f'--hparam_start_index {FLAGS.hparam_start_index} '
+  if FLAGS.hparam_end_index:
+    hparam_end_index_flag = f'--hparam_end_index {FLAGS.hparam_end_index} '
+  study_start_index = FLAGS.study_start_index if FLAGS.study_start_index else 0
+  study_end_index = FLAGS.study_end_index if FLAGS.study_end_index else num_studies - 1
+  submission_id = FLAGS.submission_id
+  rng_seed = FLAGS.seed
+
+  if not rng_seed:
+    rng_seed = struct.unpack('I', os.urandom(4))[0]
+
+  logging.info('Using RNG seed %d', rng_seed)
+  rng_key = (prng.fold_in(prng.PRNGKey(rng_seed), submission_id))
+
+  workloads = [w for w in WORKLOADS.keys()]
+
+  # Read held-out workloads
+  if FLAGS.held_out_workloads_config_path:
+    held_out_workloads = read_held_out_workloads(
+        FLAGS.held_out_workloads_config_path)
+    workloads = workloads + held_out_workloads
+
+  rng_subkeys = prng.split(rng_key, num_studies)[:num_studies:]
```

Comment:
double check why index with num_studies

Reply:
upon inspection I'm going to conclude that this must have been left-over code from a previous implementation of mine.